django/db/models/sql/query.py

```diff
@@ -1099,24 +1092,16 @@ def try_transform(self, lhs, name, rest_of_lookups):
                 "permitted." %
                 (name, lhs.output_field.__class__.__name__))
 
-    def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
-                     can_reuse=None, connector=AND, allow_joins=True, split_subq=True):
+    def build_filter(self, filter_expr, current_negated=False, can_reuse=None, allow_joins=True):
         """
         Builds a WhereNode for a single filter clause, but doesn't add it
         to this Query. Query.add_q() will then add this filter to the where
         Node.
 
-        The 'branch_negated' tells us if the current branch contains any
-        negations. This will be used to determine if subqueries are needed.
-
         The 'current_negated' is used to determine if the current filter is
         negated or not and this will be used to determine if IS NULL filtering
         is needed.
 
-        The difference between current_netageted and branch_negated is that
-        branch_negated is set on first negation, but current_negated is
-        flipped for each negation.
-
         Note that add_filter will not do any negating itself, that is done
         upper in the code by add_q().
```

Comment:
This was noted in another PR that I closed to avoid creating a merge conflict: "upper" -> "earlier"